components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1119,15 +1204,18 @@ private Condition createConditionForSingleAttributeFilter(String domainName, Nod
     /**
      * Get maximum user limit to retrieve.
      *
+     * @param domainName Name of the user store
      * @return
      */
-    private int getMaxLimit() {
+    private int getMaxLimit(String domainName) {
 
         int givenMax;
-
+        if (StringUtils.isEmpty(domainName)) {
+            domainName = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+        }
         try {
-            givenMax = Integer.parseInt(carbonUM.getRealmConfiguration().getUserStoreProperty(
-                    "MaxUserNameListLength"));
+            givenMax = Integer.parseInt(carbonUM.getSecondaryUserStoreManager(domainName).getRealmConfiguration()
+                    .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_MAX_USER_LIST));
         } catch (Exception e) {
```

Comment:
Can we use a generic exception rather catching the Exception

Reply:
Fixed